src/track/cueinfo.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+// cueinfo.h
+// Created 2020-02-28 by Jan Holthuis
+
+#include "util/color/color.h"
+
+// DTO for Cue information without dependencies on the actual Track object
+class CueInfo : public QObject {
+    Q_OBJECT
+
+  public:
+    enum class Type {
+        Invalid = 0,
+        HotCue = 1,
+        MainCue = 2,
+        Beat = 3, // unused (what is this for?)
+        Loop = 4,
+        Jump = 5,
+        Intro = 6,
+        Outro = 7,
+        AudibleSound = 8, // range that covers beginning and end of audible
+                          // sound; not shown to user
+    };
+
+    static constexpr double kNoPosition = -1.0;
+    static const int kNoHotCue = -1;
+
+    CueInfo();
+    CueInfo(CueInfo::Type type,
+            double startPositionMilliseconds,
+            double endPositionMilliseconds,
+            int hotCueNumber,
+            QString label,
+            PredefinedColorPointer color);
+
+    CueInfo::Type getType() const;
+    void setType(CueInfo::Type type);
+
+    void setStartPosition(double positionMilliseconds);
```

Comment:
Are these function names okay or should I rename them to `setStartPositionMillis` or  `setStartPositionMsecs` to make clear that we're not using the sample position here?

Reply:
I would prefer the "Millis" suffix in the name, because there is no type-safety. The name of the parameter doesn't matter for the caller.